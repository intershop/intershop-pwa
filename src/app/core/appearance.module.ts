import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
