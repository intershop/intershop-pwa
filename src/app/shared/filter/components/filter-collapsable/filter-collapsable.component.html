<div class="filter-group" [ngClass]="{ collapsed: isCollapsed }">
  <h3 (click)="toggle()" [attr.aria-expanded]="!isCollapsed" aria-controls="collapseFilter">
    {{ title }}
    <fa-icon [icon]="['fas', isCollapsed ? 'angle-right' : 'angle-down']" class="float-right"></fa-icon>
  </h3>

  <ng-content></ng-content>
</div>
