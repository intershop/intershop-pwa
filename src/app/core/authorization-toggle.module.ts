import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
