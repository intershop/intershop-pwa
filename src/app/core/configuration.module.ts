import { NgModule } from '@angular/core';

import { SPECIAL_HTTP_ERROR_HANDLER } from './interceptors/icm-error-mapper.interceptor';
import { createPaymentErrorHandler } from './utils/http-error/create-payment.error-handler';
import { editPasswordErrorHandler } from './utils/http-error/edit-password.error-handler';
import { LoginUserErrorHandler } from './utils/http-error/login-user.error-handler';
import { requestReminderErrorHandler } from './utils/http-error/request-reminder.error-handler';
import { updatePasswordErrorHandler } from './utils/http-error/update-password.error-handler';

@NgModule({
  providers: [
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: updatePasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useClass: LoginUserErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: requestReminderErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: editPasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: createPaymentErrorHandler, multi: true },
  ],
})
export class ConfigurationModule {}
