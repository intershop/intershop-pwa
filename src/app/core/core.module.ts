import { HTTP_INTERCEPTORS, HttpClient, HttpClientModule } from '@angular/common/http';
import { NgModule, Optional, SkipSelf } from '@angular/core';
import { RouterModule } from '@angular/router';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { FormlyModule } from '@ngx-formly/core';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ReactiveComponentLoaderModule } from '@wishtack/reactive-component-loader';
import { NgxCookieBannerModule } from 'ngx-cookie-banner';
import { SWIPER_CONFIG, SwiperConfigInterface } from 'ngx-swiper-wrapper';
import { ToastrModule } from 'ngx-toastr';
import { BrowserCookiesModule } from 'ngx-utils-cookies-port';

import { ConfigurationModule } from './configuration.module';
import { ExtrasModule } from './extras.module';
import { IconModule } from './icon.module';
import { AuthInterceptor } from './interceptors/auth.interceptor';
import { MockInterceptor } from './interceptors/mock.interceptor';
import { StateManagementModule } from './state-management.module';

export function translateFactory(http: HttpClient) {
  return new TranslateHttpLoader(http, 'assets/i18n/', '.json');
}

const DEFAULT_SWIPER_CONFIG: SwiperConfigInterface = {
  direction: 'horizontal',
  keyboard: true,
  mousewheel: false,
  navigation: true,
  scrollbar: false,
};
@NgModule({
  imports: [
    BrowserCookiesModule.forRoot(),
    ConfigurationModule,
    ExtrasModule,
    FormlyModule.forRoot(),
    HttpClientModule,
    IconModule,
    NgxCookieBannerModule.forRoot({
      cookieName: 'cookieLawSeen',
    }),
    ReactiveComponentLoaderModule.forRoot(),
    RouterModule,
    StateManagementModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: translateFactory,
        deps: [HttpClient],
      },
    }),
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: MockInterceptor, multi: true },
    { provide: SWIPER_CONFIG, useValue: DEFAULT_SWIPER_CONFIG },
  ],
  // exports needed to use the cookie banner in the AppComponent
  exports: [NgxCookieBannerModule, TranslateModule],
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule,
    popoverConfig: NgbPopoverConfig
  ) {
    if (parentModule) {
      throw new Error('CoreModule is already loaded. Import it in the AppModule only');
    }

    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';
  }
}
