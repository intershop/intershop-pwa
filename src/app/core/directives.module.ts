import { NgModule } from '@angular/core';

import { ClickOutsideDirective } from './directives/click-outside.directive';
import { IdentityProviderCapabilityDirective } from './directives/identity-provider-capability.directive';
import { IntersectionObserverDirective } from './directives/intersection-observer.directive';
import { ProductContextAccessDirective } from './directives/product-context-access.directive';
import { ProductContextDirective } from './directives/product-context.directive';
import { ServerHtmlDirective } from './directives/server-html.directive';

@NgModule({
  declarations: [
    ClickOutsideDirective,
    IdentityProviderCapabilityDirective,
    IntersectionObserverDirective,
    ProductContextAccessDirective,
    ProductContextDirective,
    ServerHtmlDirective,
  ],
  exports: [
    ClickOutsideDirective,
    IdentityProviderCapabilityDirective,
    IntersectionObserverDirective,
    ProductContextAccessDirective,
    ProductContextDirective,
    ServerHtmlDirective,
  ],
})
export class DirectivesModule {}
