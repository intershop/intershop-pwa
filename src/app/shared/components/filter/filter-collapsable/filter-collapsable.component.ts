import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

@Component({
  selector: 'ish-filter-collapsable',
  templateUrl: './filter-collapsable.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class FilterCollapsableComponent implements OnInit {
  @Input() title: string;
  @Input() collapsed: boolean;
  @Output() collapsedChange = new EventEmitter<boolean>();

  isCollapsed = false;

  ngOnInit() {
    this.isCollapsed = this.collapsed;
  }

  toggle() {
    this.isCollapsed = !this.isCollapsed;
    this.collapsedChange.emit(this.isCollapsed);
  }
}
