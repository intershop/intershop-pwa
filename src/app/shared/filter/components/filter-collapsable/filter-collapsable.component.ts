import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

@Component({
  selector: 'ish-filter-collapsable',
  templateUrl: './filter-collapsable.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class FilterCollapsableComponent implements OnInit {
  @Input() title: string;
  @Input() collapsed: boolean;
  @Output() collapsedChanged = new EventEmitter<boolean>();

  isCollapsed = false;

  ngOnInit() {
    this.isCollapsed = this.collapsed;
  }

  toggle() {
    this.isCollapsed = !this.isCollapsed;
    this.collapsedChanged.emit(this.isCollapsed);
  }
}
