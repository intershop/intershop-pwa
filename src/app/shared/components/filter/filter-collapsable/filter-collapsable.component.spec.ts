import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
import { FaIconComponent } from '@fortawesome/angular-fontawesome';
import { MockComponent } from 'ng-mocks';

import { FilterCollapsableComponent } from './filter-collapsable.component';

describe('Filter Collapsable Component', () => {
  let component: FilterCollapsableComponent;
  let fixture: ComponentFixture<FilterCollapsableComponent>;
  let element: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FilterCollapsableComponent, MockComponent(FaIconComponent)],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FilterCollapsableComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
  });

  it('should hide content when header is clicked', fakeAsync(() => {
    fixture.detectChanges();
    expect(element).toMatchSnapshot('open');
    const filterGroupHead = fixture.nativeElement.querySelectorAll('h3')[0];
    filterGroupHead.click();
    tick(500);
    fixture.detectChanges();

    expect(element).toMatchSnapshot('closed');
  }));
});
