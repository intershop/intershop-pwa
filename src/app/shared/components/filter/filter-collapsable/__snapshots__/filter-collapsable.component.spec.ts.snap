// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Collapsable Component should hide content when header is clicked: closed 1`] = `
<div class="filter-group collapsed">
  <h3 aria-controls="collapseFilter" aria-expanded="false">
    <fa-icon class="float-right" ng-reflect-icon="fas,angle-right"></fa-icon>
  </h3>
</div>
`;

exports[`Filter Collapsable Component should hide content when header is clicked: open 1`] = `
<div class="filter-group">
  <h3 aria-controls="collapseFilter" aria-expanded="true">
    <fa-icon class="float-right" ng-reflect-icon="fas,angle-down"></fa-icon>
  </h3>
</div>
`;
